src/lib/components/table/table-data-source.ts

```diff
@@ -0,0 +1,154 @@
+import {
+    CollectionViewer,
+    DataSource
+} from '@angular/cdk/collections';
+import {
+    EMPTY,
+    merge,
+    Observable,
+    Subject
+} from 'rxjs';
+import {
+    debounceTime,
+    map,
+    switchMap,
+    takeUntil,
+    tap
+} from 'rxjs/operators';
+import { EventEmitter } from '@angular/core';
+import { Sort } from '@angular/material/sort';
+import {
+    PageEvent
+} from '@angular/material/paginator';
+import { TerraFilter } from './filter';
+import { TerraPagerInterface } from '../pager/data/terra-pager.interface';
+import { HasPaginatorInterface } from './has-paginator.interface';
+
+/**
+ * @description Data Source base class for a data table.
+ */
+export abstract class TableDataSource<T> extends DataSource<T>
+{
+    /**
+     * @description The data to display in the table.
+     */
+    public data:Array<T> = [];
+
+    /**
+     * @description The filter instance
+     */
+    public filter:TerraFilter<unknown>;
+
+    /**
+     * @description Stream to finish all events.
+     */
+    private _disconnect$:Subject<void> = new Subject();
+
+    /**
+     * @description Connects a collection viewer (such as a data-table) to this data source. Note that
+     * the stream provided will be accessed during change detection and should not directly change
+     * values that are bound in template views.
+     * @param collectionViewer The component that exposes a view over the data provided by this
+     *     data source.
+     * @returns Observable that emits a new value when the data changes.
+     */
```

Comment:
Können wir hier vielleicht eine konkretere Beschreibung aufnehmen. Das scheint von der Basisklasse kopiert worden zu sein, in der sie abstract ist. Wir implementieren die Methode. D.h. sie tut etwas konkretes, und ist nichtmehr abstract.

Reply:
ja das stimmt.